spec/controllers/spree/admin/reports/distributor_totals_by_supplier_spec.rb

```diff
@@ -0,0 +1,249 @@
+require 'spec_helper'
+
+describe Spree::Admin::ReportsController, type: :controller do
```

Comment:
I tried to mimic the report I generated with sample data in staging. All of it without shortcuts so I managed to control all the data involved, which enabled me to understands all the bits and pieces of the report generation. I don't plan to keep after I finally close the issue. Meanwhile, it provides the lab conditions necessary to iterate quickly.